packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FGMGroundOverlayController.m

```diff
@@ -0,0 +1,234 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FGMGroundOverlayController.h"
+
+#import "FGMImageUtils.h"
+#import "FLTGoogleMapJSONConversions.h"
+
+@interface FGMGroundOverlayController ()
+
+/// The GMSMapView to which the ground overlays are added.
+@property(weak, nonatomic) GMSMapView *mapView;
+
+@end
+
+@implementation FGMGroundOverlayController
+
+- (instancetype)initWithGroundOverlay:(GMSGroundOverlay *)groundOverlay
+                           identifier:(NSString *)identifier
+                              mapView:(GMSMapView *)mapView
+                  isCreatedWithBounds:(BOOL)isCreatedWithBounds {
+  self = [super init];
+  if (self) {
+    _groundOverlay = groundOverlay;
+    _mapView = mapView;
+    _groundOverlay.userData = @[ identifier ];
+    _createdWithBounds = isCreatedWithBounds;
+  }
+  return self;
+}
+
+- (void)removeGroundOverlay {
+  self.groundOverlay.map = nil;
+}
+
+- (void)setConsumeTapEvents:(BOOL)consumes {
+  self.groundOverlay.tappable = consumes;
+}
+
+- (void)setVisible:(BOOL)visible {
+  self.groundOverlay.map = visible ? self.mapView : nil;
+}
+
+- (void)setZIndex:(int)zIndex {
+  self.groundOverlay.zIndex = zIndex;
+}
+
+- (void)setAnchor:(CGPoint)anchor {
+  self.groundOverlay.anchor = anchor;
+}
+
+- (void)setBearing:(CLLocationDirection)bearing {
+  self.groundOverlay.bearing = bearing;
+}
+
+- (void)setTransparency:(float)transparency {
+  float opacity = 1.0 - transparency;
+  self.groundOverlay.opacity = opacity;
+}
+
+- (void)setPositionFromBounds:(GMSCoordinateBounds *)bounds {
+  self.groundOverlay.bounds = bounds;
+}
+
+- (void)setPositionFromCoordinates:(CLLocationCoordinate2D)coordinates {
+  self.groundOverlay.position = coordinates;
+}
+
+- (void)setIcon:(UIImage *)icon {
+  self.groundOverlay.icon = icon;
+}
+
+- (void)updateFromPlatformGroundOverlay:(FGMPlatformGroundOverlay *)groundOverlay
+                              registrar:(NSObject<FlutterPluginRegistrar> *)registrar
+                            screenScale:(CGFloat)screenScale {
+  [self setConsumeTapEvents:groundOverlay.clickable];
+  [self setVisible:groundOverlay.visible];
```

Comment:
Based on investigation of similar methods in a recent PR (which hasn't landed), this needs to be last to avoid potential flickering (likely due to Google Maps using its own render thread), so should be at the end of the method, and have a comment saying that nothing should be added after it.